lib/registration.pm

```diff
@@ -57,18 +57,25 @@ our %SLE15_DEFAULT_MODULES = (
 sub fill_in_registration_data {
     my ($addon, $uc_addon);
     if (!get_var("HDD_SCC_REGISTERED")) {
-        # SLE 15 & textmode
-        if (sle_version_at_least('15') && check_var('DESKTOP', 'textmode')) {
-            send_key "alt-m";    # select email field if yast2 add-on
+        if (!get_var("SMT_URL")) {
+            # SLE 15 & textmode
+            if (sle_version_at_least('15') && check_var('DESKTOP', 'textmode')) {
+                send_key "alt-m";    # select email field if yast2 add-on
+            }
+            else {
+                send_key "alt-e";    # select email field if installation
+            }
+            send_key "backspace";    # delete m or e
+            type_string get_required_var('SCC_EMAIL') if get_var('SCC_EMAIL');
+            save_screenshot;         # show typed value or empty fields also as synchronization
+            send_key "alt-c";        # select registration code field
+            type_string get_required_var('SCC_REGCODE') if get_var('SCC_REGCODE');
         }
         else {
-            send_key "alt-e";    # select email field if installation
+            send_key "alt-i";
+            send_key "alt-o";
+            type_string get_required_var("SMT_URL");
         }
-        send_key "backspace";    # delete m or e
-        type_string get_required_var('SCC_EMAIL') if get_var('SCC_EMAIL');
-        save_screenshot;         # show typed value or empty fields also as synchronization
-        send_key "alt-c";        # select registration code field
-        type_string get_required_var('SCC_REGCODE') if get_var('SCC_REGCODE');
```

Comment:
I remember that if SMT_URL is set and it would be appended to Boot Options, then Installer would use it by default to do registration no matter what the preferred method you selected in 'Registration' screen. So above changes are not very necessary from my opinion, and without any change on current lib/registration.pm, you can register via smt without problem.